app/controllers/api/subcollections/cloud_volume_snapshots.rb

```diff
@@ -0,0 +1,57 @@
+module Api
+  module Subcollections
+    module CloudVolumeSnapshots
+      def cloud_volume_snapshots_query_resource(object)
+        object.cloud_volume_snapshots
+      end
+
+      def cloud_volume_snapshots_create_resource(parent, _type, _id, data)
+        raise parent.unsupported_reason(:snapshot_create) unless parent.supports?(:snapshot_create)
```

Comment:
Should this be using the new refactorings?

Reply:
We have the non sub-collections case nailed down.
This cases are so similar, but there are slight changes around the model passed to supports and the way the resources were looked up.

It is similar enough that creating a separate method feels wrong.
They are different enough that having a method support both got ugly fast

For now I was just punting, but it does look like a few methods could be simplified.

Reply:
Sorry I didn't share with you sooner.